KarhooUISDK/Screens/TripScreen/KarhooTripPresenter.swift

```diff
@@ -185,7 +186,17 @@ final class KarhooTripPresenter: TripPresenter,
         cameraShouldFollowCar = true
 
         if trip.state == .driverEnRoute || trip.state == .arrived {
-            tripView?.focusOnUserLocation()
+            var isLocationPermissionGranted: Bool {
+                CLLocationManager.authorizationStatus() == .authorizedWhenInUse ||
+                    CLLocationManager.authorizationStatus() == .authorizedAlways
+            }
+            switch isLocationPermissionGranted {
+            case true:
+                tripView?.focusOnUserLocation()
+            case false:
+                tripView?.showNoLocationPermissionsPopUp()
```

Comment:
This is called every time the trip status is updated, which is every few seconds (even if it's to notify that the status hasn't changed, we still get a ping), which leads to the user seeing the alerts stack one on top of each other. Let's show this alert only when the user taps the location button. Otherwise just focus on pickup / driver depending of trip state